### PR TITLE
feat : add loop to display result of all action types

### PR DESCRIPTION
**Pull request by chiragw15**
Fixes issue #664 Part 1

Changes: Added loop to display result of each action type

### CODE REVIEW

**Comment by chiragw15** [comment]
@mayank408 @amitiwary999 can you guys please test this. Pull the code, run it on you device or emulator and test if everything works fine. What I have done in this : 
Commit 1. Added a loop and a delay handler to parse actions and display them one after another with the delay provided.
Commit 2, Part 1. Changed view holders to align them with the new way of display each action response in a different message.
Commit 2, Part 2 : Changed the way of updating database. Used to always find out the last message in database before entering new message. Just changed it by making a global variable, assigning it with index of last message in database in the starting and then just incrementing the variable, every time a new database entry is made. 
Commit 3, Part 1 : Changed implementation of parsing of json. Like there were boolean variables like ismap, iswebsearch, isrss, ispiechart etc. Remove all of them and added a String variable called actionType which stores action type. This also reduced the number of parameters in updateDatabase() method and number of variables in ChatMessage class.
Commit 3, Part 1 : Fixed map action type by introducing a new model for map which stores latitude, longitude and zoom.
Commit 4 : Updated implementation of retrieving old messages. Simply added a loop.
Commit 5 : Changed delete, share and copy options to adapt to new implementation.

There will be small bugs here and there, just report them here. Or report if you find a better way to implement something. Need to test the app thoroughly before sending final PR. So, report as many bugs or poor implementation as you can.

File: app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java [hunk 1]
```diff
@@ -269,62 +267,75 @@ public boolean checkMicInput() {
...
-        try {
-            isWebSearch = susiResponse.getAnswers().get(0).getActions().get(i).getType().equals("websearch");
-            datumList = susiResponse.getAnswers().get(0).getData();
-            webSearch = susiResponse.getAnswers().get(0).getActions().get(1).getQuery();
-        } catch (Exception e) {
-            isWebSearch = false;
-            webSearch = "";
-        }
+            case Constant.MAP :
+                try {
```
Comment (amitiwary999) [inline app/src/main/java/org/fossasia/susi/ai/activities/MainActivity.java]: Here you used the same thing which I used to solve this problem but you rejected that and now you used same thing.You even not tell me about that.  I am working on this today. Please let me solve some issues. For your kind information I am also part of gsoc 2017 program. @mariobehling can you please see it.

**Comment by amitiwary999** [comment]
@chiragw15 why you are trying to solve every problem in one pull and you even not mention it. Please I am again requesting let me solve my issues. Please.

**Comment by chiragw15** [comment]
@amitiwary999 Relax. I am not solving your issues. You have already solved it and sent a PR. Why would I solve it again? I just copied the code from your PR and used it here. PR #683 and #668 .This is for testing if everything will work fine if your code and my code merges together. Relax. your code will be removed. I hope that clears the misunderstanding. Can you test it once? Although I tested but more testers will be better. Once everything works fine, I'll remove the code I copied from your PR and send update the PR. Cheers man. I really respect your work and I am not trying to steal it away. :smile:

**Comment by chiragw15** [comment]
Just to remove the confusion, there is some extra code taken from other PR's to test if code in this PR aligns with them or not. I will remove that code once the app is tested completely. I will then make a final update of PR which will contain only code solving issue #664 and ask for review. Hope This clears things.

**Comment by amitiwary999** [comment]
I test it everything looks fine but tts is very slow and sometimes susi is not able convert text to speech . I will test it on more devices :+1:

**Comment by chiragw15** [comment]
@amitiwary999 check now. I added TODO statements where your code will be added. You may have to resolve merge conflicts for your open PRs . Just remove the TODO statements while resolving conflicts.
@mayank408 @betterclever can you review the PR please. For now anchor action type won't work. It will work after merging of #683 and #689 .

**Comment by mayank408** [comment]
@chiragw15 Thoroughly tested the app, looking fine if got an error after will report it. Cheers

**Comment by chiragw15** [comment]
The updation of map action type is done by @amitiwary999  in PR #668, I merged his PR into mine to test if everything aligns with the new loop implementation. It needed some tweaks. Did that but now it will be very complex and will be time wasting to remove them and add them again. So, I am leaving them as it is.

**Pull request merged**